docs/getting-started-guides/docker-multinode/functions.sh

```diff
@@ -0,0 +1,343 @@
+      echo >&2 "Please run as root"
+      exit 1
+    fi
+    # Get current ip
+    CURRENT_IP=`ifconfig $FLANNEL_IF | grep 'inet addr:' | cut -d: -f2 | awk '{ print $1}'`
+    if [ -z ${CURRENT_IP} ]; then
+      CURRENT_IP=`hostname -i|xargs`
+    fi
+    # Make sure master ip is properly set
+    if [ -z ${MASTER_IP} ]; then
+        echo "Please export MASTER_IP in your env"
+        exit 1
+    fi
+    echo "K8S_VERSION is set to: ${K8S_VERSION}"
+    echo "ETCD_VERSION is set to: ${ETCD_VERSION}"
+    echo "FLANNEL_VERSION is set to: ${FLANNEL_VERSION}"
+    echo "FLANNEL_IFACE is set to: ${FLANNEL_IFACE}"
+    echo "MASTER_IP is set to: ${MASTER_IP}"
+    echo "ARCH is set to: ${ARCH}"
+}
+
+# Detect the OS distro, we support ubuntu, debian, mint, centos, fedora dist
+detect_lsb() {
+    case "$(uname -m)" in
+        *64)
+            ;;
+         *)
+            echo "Error: We currently only support 64-bit platforms."
+            exit 1
+            ;;
+    esac
+
+    if command_exists lsb_release; then
+        lsb_dist="$(lsb_release -si)"
+    fi
+    if [ -z ${lsb_dist} ] && [ -r /etc/lsb-release ]; then
+        lsb_dist="$(. /etc/lsb-release && echo "$DISTRIB_ID")"
+    fi
+    if [ -z ${lsb_dist} ] && [ -r /etc/debian_version ]; then
+        lsb_dist='debian'
+    fi
+    if [ -z ${lsb_dist} ] && [ -r /etc/fedora-release ]; then
+        lsb_dist='fedora'
+    fi
+    if [ -z ${lsb_dist} ] && [ -r /etc/os-release ]; then
+        lsb_dist="$(. /etc/os-release && echo "$ID")"
+    fi
+
+    lsb_dist="$(echo ${lsb_dist} | tr '[:upper:]' '[:lower:]')"
+
+    case "${lsb_dist}" in
+        amzn|centos|debian|ubuntu)
+            ;;
+        *)
+            echo "Error: We currently only support ubuntu|debian|amzn|centos."
+            exit 1
+            ;;
+    esac
+}
+
+etcd() {
+  if docker_bootstrap_run etcd; then
+    return 0
+  fi
+  if local_master; then
+    echo "Launch new etcd cluster"
+    etcd_master
+  else
+    etcd_join
+  fi
+}
+
+etcd_master() {
+   docker-bootstrap run \
+      --restart=always \
+      --net=host \
+      --name=etcd \
+      -d \
+      gcr.io/google_containers/etcd:${ETCD_VERSION} \
+      /usr/local/bin/etcd \
+      --listen-client-urls=http://0.0.0.0:2379,http://0.0.0.0:4001 \
+      --advertise-client-urls=http://${MASTER_IP}:2379,http://${MASTER_IP}:4001 \
+      --data-dir=/var/etcd/data \
+      --initial-advertise-peer-urls http://${MASTER_IP}:2380 \
+      --listen-peer-urls http://0.0.0.0:2380 \
+      --initial-cluster-token `hostname` \
+      --initial-cluster default=http://${MASTER_IP}:2380 \
+      --initial-cluster-state new
+
+      sleep 5
+      # Set flannel net config
+      docker -H unix:///var/run/docker-bootstrap.sock run \
+      --net=host gcr.io/google_containers/etcd:${ETCD_VERSION} \
+      etcdctl \
+      set /coreos.com/network/config \
+      '{ "Network": "10.1.0.0/16", "Backend": {"Type": "vxlan"}}'
+}
+
+etcd_join() {
```

Comment:
Why would a worker join the `etcd` cluster? Only flannel should be started in docker-bootstrap on a worker